From linux-usb-owner@vger.kernel.org  Tue Sep 21 13:43:34 2010
From: Keshava Munegowda <keshava_mgowda@ti.com>
To: linux-usb@vger.kernel.org, linux-omap@vger.kernel.org
Cc: gadiyar@ti.com, nskamat@ti.com,
	Keshava Munegowda <keshava_mgowda@ti.com>
Subject: usb: omap: ohci: Missing driver unregister in module exit
Date: Tue, 14 Sep 2010 04:40:01 +0530
Message-Id: <1284419401-32101-1-git-send-email-keshava_mgowda@ti.com>

The un-registration of OHCI driver was not done in the ohci_hcd_mod_exit
function. This was affecting rmmod command not to work for OMAP3
platforms.  The platform driver un-registration for OMAP3 platforms is
perfomed while removing the OHCI module from kernel.

Signed-off-by: Keshava Munegowda <keshava_mgowda@ti.com>
Signed-of-by: Felipe Balbi <balbi@ti.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci-hcd.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/usb/host/ohci-hcd.c
+++ b/drivers/usb/host/ohci-hcd.c
@@ -1277,6 +1277,9 @@ static void __exit ohci_hcd_mod_exit(voi
 #ifdef PLATFORM_DRIVER
 	platform_driver_unregister(&PLATFORM_DRIVER);
 #endif
+#ifdef OMAP3_PLATFORM_DRIVER
+	platform_driver_unregister(&OMAP3_PLATFORM_DRIVER);
+#endif
 #ifdef PS3_SYSTEM_BUS_DRIVER
 	ps3_ohci_driver_unregister(&PS3_SYSTEM_BUS_DRIVER);
 #endif
